### PR TITLE
fix: Stream monitor behavior on failed and stalled upload attempts

### PR DESCRIPTION
**Pull request by cacieprins**
- Closes 

### Additional details
- When the stream monitor aborts an upload due to a stall, the monitor's error is raised rather than the more general "User aborted the request" error message
- The stream monitor no longer short-circuits certain operating system connection timeouts, which will result in more precise error messages when connections cannot be established.

### PR Tasks

- [X] Have tests been added/updated?
- [NA] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [NA] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: cli/CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,12 @@
 <!-- See the ../guides/writing-the-cypress-changelog.md for details on writing the changelog. -->
+## 13.13.3
+
+_Released 8/13/2024_
```
Comment (cacieprins) [inline cli/CHANGELOG.md]:
```suggestion
_Released 8/13/2024 (PENDING)_
```

File: packages/server/lib/cloud/api/put_protocol_artifact.ts [hunk 2]
```diff
@@ -25,23 +23,27 @@ export const putProtocolArtifact = asyncRetry(
...
+          'accept': 'application/json',
+        },
+        // ts thinks this is a web fetch, which only expects ReadableStreams.
+        // But, this is a node fetch, which supports ReadStreams.
+        // @ts-expect-error
+        body: activityMonitor.monitor(fileStream),
+        signal: controller.signal,
+      })
+    } catch (e) {
+      throw e
```
Comment (AtofStryker) [inline packages/server/lib/cloud/api/put_protocol_artifact.ts]: whats the point of catching to immediately rethrow?

File: packages/server/test/unit/cloud/network/put_fetch_spec.ts [hunk 3]
```diff
@@ -107,22 +107,65 @@ describe('cloud/network/put_fetch', () => {
...
         await fetch(url, { parse: 'text' })
       } catch (e) {
         err = e
       }
-      expect(NetworkError.isNetworkError(err)).to.be.true
+      expect(SystemError.isSystemError(err)).to.be.true
+    })
+  })
+
+  describe('when fetch is provided an abort signal, and rejects via signal', () => {
```
Comment (AtofStryker) [inline packages/server/test/unit/cloud/network/put_fetch_spec.ts]:
```suggestion
  describe('when fetch is provided with an abort signal, and rejects via signal', () => {
```

File: packages/server/lib/cloud/api/put_protocol_artifact.ts (hunk 2 above)
Comment (cacieprins) [reply packages/server/lib/cloud/api/put_protocol_artifact.ts]: I had some temporary debugging in there - will remove

Comment (cacieprins) [reply packages/server/lib/cloud/api/put_protocol_artifact.ts]: addressed in https://github.com/cypress-io/cypress/pull/29959/commits/d302a052dacd5b05322bff32737992da28022f2f

**Review by mschile — APPROVED** [review]

**Review by ryanthemanuel — APPROVED** [review]

**Pull request merged**